File: 백준/Silver/1931. 회의실 배정/회의실 배정.js
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8').toString().trim().split('\n');
const N = input[0];
const schedules = input.slice(1).map(x => x.split(' ').map(Number)).sort((a, b) => a[1] === b[1] ? a[0] - b[0] : a[1] - b[1]);
let lastEndTime = 0;
let answer = 0;

for (let [start, end] of schedules) {
    if (start < lastEndTime) continue;
    
    lastEndTime = end;
    answer++;
}

console.log(answer);

File: 백준/Gold/1931. 회의실 배정/회의실 배정.py
import sys

N = int(input())
M = [list(map(int, row.split())) for row in sys.stdin.readlines()]
M.sort(key = lambda x: (x[1], x[0]))
t = 0
c = 0

for s, e in M:
    if t <= s:
        c += 1
        t = e
        
print(c)